From 763a046849ded07e3785d76b8af24381b44aadc1 Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Sun, 13 Jun 2021 12:13:30 +0100
Subject: [PATCH 1/2] alpm - fix uninitialized stack vars used later in config

In the parsing of siglevel, depending on content this can leave
level and mask sometimes unset as the code likes to take current value
and val | newbitval to set bits, assuming it starts at 0. This fixes
that.
---
 backends/alpm/pk-alpm-config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backends/alpm/pk-alpm-config.c b/backends/alpm/pk-alpm-config.c
index 01081106c..3fbb60c18 100644
--- a/backends/alpm/pk-alpm-config.c
+++ b/backends/alpm/pk-alpm-config.c
@@ -793,7 +793,7 @@ static gboolean
 pk_alpm_config_configure_repos (PkBackend *backend, PkAlpmConfig *config,
 				   alpm_handle_t *handle, GError **error)
 {
-	alpm_siglevel_t base, level, mask, local, remote;
+	alpm_siglevel_t base, level = 0, mask = 0, local, remote;
 	const alpm_list_t *i;
 	PkAlpmConfigSection *options;
 
-- 
2.31.1

